src/aggregation/metric/top_hits.rs

```diff
@@ -0,0 +1,820 @@
+            .into_iter()
+            .map(|field| {
+                if !field.contains('*') && reader.iter_columns()?.any(|(name, _)| name == field) {
+                    return Ok(vec![field]);
+                }
+
+                let pattern = globbed_string_to_regex(&field)?;
+                Ok(reader
+                    .iter_columns()?
+                    .filter(|(name, _)| pattern.is_match(name))
+                    .map(|(name, _)| name)
+                    .collect::<Vec<_>>())
+            })
+            .collect::<crate::Result<Vec<_>>>()?
+            .into_iter()
+            .flatten()
+            .collect();
+
+        Ok(Self {
+            doc_value_fields: resolved_fields,
+        })
+    }
+
+    fn get_fields(
+        &self,
+        accessors: &HashMap<String, DynamicColumn>,
+        doc_id: DocId,
+    ) -> FieldRetrivalResult {
+        let dvf = self
+            .doc_value_fields
+            .iter()
+            .map(|field| {
+                let accessor = accessors.get(field).expect("field accessor must exist");
+
+                let value: OwnedValue = match accessor {
+                    DynamicColumn::U64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::U64),
+                    DynamicColumn::I64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::I64),
+                    DynamicColumn::F64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::F64),
+                    DynamicColumn::Bytes(accessor) => 'l: {
+                        let Some(term_ord) = accessor.term_ords(doc_id).next() else {
+                            break 'l OwnedValue::Null;
+                        };
+                        let mut buffer = vec![];
+                        match accessor
+                            .ord_to_bytes(term_ord, &mut buffer)
+                            .expect("term dictionary read must succeed")
+                        {
+                            false => OwnedValue::Null,
+                            true => OwnedValue::Bytes(buffer),
+                        }
+                    }
+                    DynamicColumn::Str(accessor) => 'l: {
+                        let Some(term_ord) = accessor.term_ords(doc_id).next() else {
+                            break 'l OwnedValue::Null;
+                        };
+                        let mut buffer = vec![];
+                        match accessor
+                            .ord_to_bytes(term_ord, &mut buffer)
+                            .expect("term dictionary read must succeed")
+                        {
+                            false => OwnedValue::Null,
```

Comment:
I think all stored term ords do exist

Reply:
I think this will return an Error on an IO failure. Should be bubbled up instead?

Reply:
I mean we never get back false, or that's an error

Reply:
Replaced with plain `.unwrap()` as elsewhere in the codebase

Reply:
> I mean we never get back false, or that's an error

Oh right, good catch!

Reply:
`.expect` is definitely better than unwrap as you can provide context

Reply:
Good point [5162e14](https://github.com/quickwit-oss/tantivy/pull/2198/commits/5162e140da94929fcea042697aaa741d116e89ed)